src/datepicker/datepicker.component.ts

```diff
@@ -211,6 +212,13 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 		this.onTouched();
 	}
 
+	@HostListener("document:scroll")
```

Comment:
This is only going to catch the document scrolling, which won't catch the modal scrolling - see here for a demo https://stackblitz.com/edit/typescript-hvdqcp?file=index.ts

You should be able to use `scrollableParentsObservable` from `utils/scroll`